config/initializers/new_framework_defaults_5_2.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 5.2 upgrade.
+#
+# Once upgraded flip defaults one by one to migrate to the new default.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+# Make Active Record use stable #cache_key alongside new #cache_version method.
+# This is needed for recyclable cache keys.
+# Rails.application.config.active_record.cache_versioning = true
+
+# Use AES-256-GCM authenticated encryption for encrypted cookies.
+# Also, embed cookie expiry in signed or encrypted cookies for increased security.
+#
+# This option is not backwards compatible with earlier Rails versions.
+# It's best enabled when your entire app is migrated and stable on 5.2.
+#
+# Existing cookies will be converted on read then written with the new scheme.
+# Rails.application.config.action_dispatch.use_authenticated_cookie_encryption = true
+
+# Use AES-256-GCM authenticated encryption as default cipher for encrypting messages
+# instead of AES-256-CBC, when use_authenticated_message_encryption is set to true.
+# Rails.application.config.active_support.use_authenticated_message_encryption = true
+
+# Add default protection from forgery to ActionController::Base instead of in
+# ApplicationController.
+# Rails.application.config.action_controller.default_protect_from_forgery = true
+
+# Store boolean values are in sqlite3 databases as 1 and 0 instead of 't' and
+# 'f' after migrating old data.
+# Rails.application.config.active_record.sqlite3.represent_boolean_as_integer = true
+
+# Use SHA-1 instead of MD5 to generate non-sensitive digests, such as the ETag header.
+# Rails.application.config.active_support.use_sha1_digests = true
+
+# Make `form_with` generate id attributes for any generated HTML tags.
+# Rails.application.config.action_view.form_with_generates_ids = true
```

Comment:
Maybe, append [this content](https://github.com/rails/rails/blob/v5.1.7/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_1.rb.tt)?

Reply:
@eitoball here the right file for the current version: https://github.com/rails/rails/blob/v5.2.4/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_2.rb.tt

Reply:
This file lists new configurations that are added in Rails 5.2. The file that I mentioned is for new defaults in Rails 5.1. If we change `config.load_defaults` in `config/application.rb` to `5.1` or `5.2`, new defaults becomes effective, and application might fail. Therefore, I think that we should append or add new file like `new_framework_defaults_5_1.rb`.

Reply:
That’s why we have specs. I tested with rails 5.2 default and it seems working. Let me know if you still think we need to use 5.1 defaults even tho this PR upgrades to 5.2

Reply:
Yes, we have specs, but specs don't find bugs. specs only make sure what you expect are working in way that you expected.

Setting `config.load_defauls` to `5.2' also set 5.1 defaults, so I think that we should put 5.1 defaults like 5.2 defaults in this file or another separate file.

Reply:
> Yes, we have specs, but specs don't find bugs. specs only make sure what you expect are working in way that you expected.

Well, specs let you catch bugs in your logic before code goes out for review/release. They can also, depending on what and how you test, help you make sure your code is working before it's in a deployable state. Part of the reason that we trust specs is that we've seen them fail. It's like [double-entry bookkeeping](https://en.wikipedia.org/wiki/Double-entry_bookkeeping). The specs keep the [SUT](https://en.wikipedia.org/wiki/System_under_test) in place, and the SUT makes the tests pass.

Anyways, here is an update 4e6e446